core/src/main/scala/com/raphtory/internals/management/QuerySender.scala

```diff
@@ -51,6 +54,13 @@ private[raphtory] class QuerySender(
     }
 
   def submit(query: Query, customJobName: String = ""): QueryProgressTracker = {
+
+    if (lastQueryIndex < individualUpdateIndex) {
+      unblockIngestion(sourceID = sourceID, individualUpdateIndex, force = false)
+      lastQueryIndex = individualUpdateIndex
+      blockingSources += sourceID
+    }
+
     val jobName     = if (customJobName.nonEmpty) customJobName else getDefaultName(query)
```

Comment:
This creates a race condition in the case of multiple sets of updates and queries. We need to integrate the unblock information into the query message itself.